### ISSUE TITLE
BusyBox/ed crashes with illegal instruction on Ovirt using OVA QCOW2 4.11

### ISSUE THREAD

**Issue by derekatkins** [home-assistant/addons#2338]
Running 'ed configuration.yaml' and I get an Illegal Instruction (core dumped).

**HassOS release with the issue:**

HassOS 4.11

**Supervisor logs:**

There is nothing in the log from the timeframe of the crash.

**Journal logs:**

config $ journalctl
-bash: journalctl: command not found

**Kernel logs:**

```
[446091.798254] traps: ed[15810] trap invalid opcode ip:55ba78f707d4 sp:7ffeaa206580 error:0 in busybox[55ba78eed000+9c000]
[446091.802066] audit: type=1701 audit(1596393071.567:119): auid=4294967295 uid=0 gid=0 ses=4294967295 subj==docker-default (enforce) pid=15810 comm="ed" exe="/bin/busybox" sig=4 res=1
```

**Description of problem:**
I am running HassOS 4.11 within my OVirt VM infrastructure.  I installed the OVA QCOW2 image and built the VM around that using the OVA configuration information (2G RAM, 2 CPUs).  The system has been up and running for a while.

I was trying to use 'ed' to modify the configuration.yaml (I hate vi, and you don't have emacs installed).  But all of a sudden it seems to crash:

```
config $ ed configuration.yaml 
"configuration.yaml", 47 lines, 1071 chars
: 2p
# Configure a default setup of Home Assistant (frontend, api, etc)
: s/#
ed: missing 2nd delimiter for substitute
: s/#//
Illegal instruction (core dumped)
```

It was not doing this when I first started using HassOS.  I do not see any issues in the VM Host logs.

**Comment by agners** [comment]
I can also reproduce this on my x86-64 machine:

```
$ docker run -it alpine 
/ # echo "# test" > test
/ # ed test
"test", 1 lines, 7 chars
: s/#//
Illegal instruction (core dumped)
```

So it probably should get reported to Alpine Linux directly: https://wiki.alpinelinux.org/wiki/Alpine_Linux:Contribute

**Comment by derekatkins** [comment]
Oh, was that meant for me to go report it to Alpine?

**Comment by agners** [comment]
@derekatkins yes, sorry, should have been more specific. I don't care about `ed` that much :slightly_smiling_face: . The check above with the official Alpine image only shows that it hasn't been caused by something on our end.

**Comment by derekatkins** [comment]
@agners Fair enough.  Done.
See https://gitlab.alpinelinux.org/alpine/aports/-/issues/13504

**Comment by derekatkins** [comment]
Looks like upstream is working on fixing it (and has actually already pushed a commit).
Not sure when it'll be in a state where this add-on can pull in the fix?

**Comment by derekatkins** [comment]
... still a bug ....

**Comment by agners** [comment]
> Looks like upstream is working on fixing it (and has actually already pushed a commit). Not sure when it'll be in a state where this add-on can pull in the fix?

It seems that the fix didn't got backported to Alpine 3.15 (https://gitlab.alpinelinux.org/alpine/aports/-/commits/3.15-stable/main/busybox). So it likely will need Alpine 3.16, and then the bump to the CLI add-on, so unfortunately it will take quite a bit.

**Comment by agners** [comment]
This is no longer reproducible. Most likely got fixed by #2576.
